### PR TITLE
remove fluid.initializer.UniformInitializer, ConstantInitializer, NormalInitializer, TruncatedNormalInitializer, XavierInitializer, BilinearInitializer, MSRAInitializer, NumpyArrayInitializer and calculate_gain..

### PR DESCRIPTION
**Pull request by zhengqiwen1997**
### PR types

Others
### PR changes

APIs
### Describe

To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.initializer.UniformInitializer, paddle.fluid.initializer.ConstantInitializer, paddle.fluid.initializer.NormalInitializer, paddle.fluid.initializer.TruncatedNormalInitializer, paddle.fluid.initializer. XavierInitializer, paddle.fluid.initializer.BilinearInitializer, paddle.fluid.initializer.MSRAInitializer and paddle.fluid.initializer. NumpyArrayInitializer, since it is not used in version 2.x, should be removed.

For unittest code where this function is used, if it is aimed to test the old API, it can be removed directly. Otherwise, the paddle.fluid.initializer.UniformInitializer should be replaced with paddle.nn.initializer.Uniform, paddle.fluid.initializer.ConstantInitializer should be replaced with paddle.nn.initializer.Constant, paddle.fluid.initializer.NormalInitializer should be replaced with paddle.nn.initializer.Normal, paddle.fluid.initializer.TruncatedNormalInitializer should be replaced with paddle.nn.initializer.TruncatedNormal, paddle.fluid.initializer.XavierInitializer should be replaced with paddle.nn.initializer.XavierUniform or XavierNormal, paddle.fluid.initializer.BilinearInitializer should be replaced with paddle.nn.initializer.BilinearI, paddle.fluid.initializer.MSRAInitializer should be replaced with paddle.nn.initializer.KaimingUniform or KaimingNormal, paddle.fluid.initializer.NumpyArrayInitializer should be replaced with paddle.nn.initializer.Assign and paddle.fluid.initializer.calculate_gain should be replaced with paddle.nn.initializer.calculate_gain.

### CODE REVIEW

File: python/paddle/nn/initializer/__init__.py [hunk 1]
```diff
@@ -51,4 +55,7 @@
     'Dirac',
     'set_global_initializer',
     'calculate_gain',
+    'UniformInitializer',
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/__init__.py]: `__all__`的内容增加意味着新增对外的公开API，这里实际上不需要做这个动作

File: python/paddle/nn/initializer/common.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/common.py]: Copyright (c) 2022

File: python/paddle/fluid/initializer.py [hunk 3]
```diff
@@ -31,16 +31,11 @@
...
     'TruncatedNormal',
     'Xavier',
     'Bilinear',
     'MSRA',
-    'ConstantInitializer',
-    'UniformInitializer',
-    'NormalInitializer',
+    'Initializer',
     'TruncatedNormalInitializer',
     'XavierInitializer',
```
Comment (zoooo0820) [inline python/paddle/fluid/initializer.py]: 剩余的几个，合并到这个PR里一起处理吧

File: python/paddle/distributed/fleet/base/distributed_strategy.py [hunk 4]
```diff
@@ -104,7 +104,7 @@ def _set_distributed_strategy(self, dist_strategy):
         self.job_info.strategy = dist_strategy
 
 
-ReduceStrategyFluid = paddle.static.BuildStrategy.ReduceStrategy
+ReduceStrategyFluid = paddle.fluid.BuildStrategy.ReduceStrategy
```
Comment (zoooo0820) [inline python/paddle/distributed/fleet/base/distributed_strategy.py]: 这个地方的改动，是因为循环依赖吗？
因为`paddle.static.BuildStrategy` 属于公开API,这个调用理论上是没有问题的

File: python/paddle/nn/initializer/common.py [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from paddle import _C_ops
+
+from ...fluid import core, framework, unique_name
+from ...fluid.data_feeder import check_variable_and_dtype
+from ...fluid.framework import _current_expected_place, in_dygraph_mode
+from ...fluid.initializer import Initializer
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/common.py]: - `Initializer `也一起迁过来
- 其余的几个类，放在对应的文件里吧
- 这个文件命名可以再斟酌下，是否改成`initializer.py`更合适一些

File: python/paddle/nn/initializer/common.py [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+        diag_step (int): Step size between two diagonal elements,
+            which is generally the width of the square matrix.
+        diag_val (float): the value of the diagonal element to be initialized,
+            default 1.0. It takes effect only if the diag_num is greater than 0.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+            import paddle.fluid as fluid
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/common.py]: 迁移后的文档，需要符合2.0的标准，示例代码需要移除fluid的相关内容，并使用公开API进行展示

File: python/paddle/distributed/fleet/base/distributed_strategy.py (hunk 4 above)
Comment (zhengqiwen1997) [reply python/paddle/distributed/fleet/base/distributed_strategy.py]: 这里经测试，改成paddle.static.BuildStrategy.ReduceStrategy后报错，说paddle里面无static，就很奇怪。该文件其他的paddle.static.XXX就没问题。

File: python/paddle/fluid/initializer.py [hunk 7]
```diff
@@ -30,1139 +30,76 @@
...
 
-    Where, `out_channels=C` and `groups=C` means this is channel-wise transposed
-    convolution. The filter shape will be (C, 1, K, K) where K is `kernel_size`,
-    This initializer will set a (K, K) interpolation kernel for every channel
-    of the filter identically. The resulting shape of the output feature map
-    will be (B, C, factor * H, factor * W). Note that the learning rate and the
-    weight decay are set to 0 in order to keep coefficient values of bilinear
-    interpolation unchanged during training.
 
+def calculate_gain(nonlinearity, param=None):
```
Comment (zoooo0820) [inline python/paddle/fluid/initializer.py]: `calculate_gain`看起来应该一起迁移过去

File: python/paddle/nn/initializer/__init__.py [hunk 8]
```diff
@@ -49,6 +59,5 @@
     'Uniform',
     'Orthogonal',
     'Dirac',
-    'set_global_initializer',
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/__init__.py]: 这个地方，虽然不迁移但是不需要移除已公开的API

File: python/paddle/nn/initializer/Bilinear.py [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+                    inputs={"X": out_var},
+                    outputs={"Out": var},
+                    attrs={"in_dtype": out_var.dtype, "out_dtype": var.dtype},
+                )
+
+            var.op = op
+            return op
+
+
+Bilinear = BilinearInitializer
```
Comment (zoooo0820) [inline python/paddle/nn/initializer/Bilinear.py]: `BilinearInitializer`的名字没有其他地方使用，看起来没有必要同时存在`BilinearInitializer `和`Bilinear `两个名字。直接统一命名为`Bilinear`吧，与目前对外公开的名称一致。

**Review by zoooo0820 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_dist_fleet_ps5.py [hunk 10]
```diff
@@ -156,7 +166,9 @@ def get_loss(cos_q_pt, cos_q_nt):
             x=nt_ss,
             size=hid_dim,
             weight_attr=fluid.ParamAttr(
-                initializer=fluid.initializer.Constant(value=0.01),
+                initializer=paddle.nn.initializer.ConstantInitializer(
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/unittests/test_dist_fleet_ps5.py]: 考虑到Initializer比较重复，对外推荐的用法是paddle.nn.initializer.Constant
https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/initializer/Constant_cn.html#constant

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by wanghuancoder — DISMISSED** [review]
LGTM

File: python/paddle/nn/initializer/assign.py [hunk 11]
```diff
@@ -12,13 +12,127 @@
...
-from ...fluid.initializer import NumpyArrayInitializer
+from ...fluid.framework import _current_expected_place, in_dygraph_mode
+from .initializer import Initializer
 
 __all__ = []
 
 
+class NumpyArrayInitializer(Initializer):
+    """Init an parameter with an numpy array
+    This op initialize the variable by numpy array.
```
Comment (Ligoml) [inline python/paddle/nn/initializer/assign.py]: 1. api和op的概念不要在文档中混用；
2. variable的概念我们现在应该是不提了，用tensor来统一，详见这条FAQ：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/faq/2.0.html#paddle2-0-lodtensor

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by wanghuancoder — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

File: python/paddle/fluid/initializer.py [hunk 12]
```diff
@@ -30,1139 +30,24 @@
...
-    'UniformInitializer',
-    'NormalInitializer',
-    'TruncatedNormalInitializer',
-    'XavierInitializer',
-    'BilinearInitializer',
-    'MSRAInitializer',
-    'NumpyArrayInitializer',
-    'set_global_initializer',
-]
+__all__ = ['set_global_initializer']
```
Comment (jeff41404) [inline python/paddle/fluid/initializer.py:33]: why only `set_global_initializer` has not been moved or cleaned?

Comment (zhengqiwen1997) [reply python/paddle/fluid/initializer.py:33]: There is a problem of circular dependency, imported by layer_helper_base. So it cannot be moved temporarily.

**Review by jeff41404 — APPROVED** [review]

**Pull request merged**